### ISSUE TITLE
Shift not displaying any torrents

### ISSUE THREAD

**Issue by Gladuin** [killemov/Shift#30]
I'm using shift with transmission 4.0.6, but when I try to add a torrent, nothing seems to happen. I'm using the latest firefox.
Nothing out of the ordinary appears in the transmission-daemon log, and the user that's running transmission-daemon can fully access the Shift folder.

Attached is a screenshot of the Firefox console:
![image](https://github.com/user-attachments/assets/45931bb5-000f-4194-aae4-c9ac0e1671e1)

Let me know if you need more information about my setup!

**Comment by killemov** [comment]
Well this is a new one. The `settings.json` file was not in question, it is a Shift problem, so the `shift.json` file just might be. Looking at the line in question from your screenshot of the developer tools the `paths` property might be missing or have an incorrect value. It should be an array of strings. See: https://github.com/killemov/Shift/blob/3be3acb2f3b15359267ba93015bee1b6c95d90ea/shift.json#L34
Could you check that? You might fix it by going to the Session -> Shift page, Scrape Paths and then Apply.

**Comment by Gladuin** [comment]
The `paths` property is exactly as you posted it in my `shift.json`. When I try to go to the session > shift page, I get the following error in my console (this is on edge and via zerotier instead of my regular setup, but I this page also didn't work when I tried it on firefox last time)

![image](https://github.com/user-attachments/assets/eb5f3764-2749-48c8-a619-065c9bdce5e7)

Here's my `shift.json`:
```json
{
  "confirmRemove": false,
  "confirmStartAll": true,
  "confirmStopAll": false,
  "confirmTrash": true,
  "contentMagic": false,
  "dark": false,
  "deleteTorrentLabelByKey": true,
  "fileLink": "/secret/${torrent.downloadDir}/",
  "fileLinkEnabled": false,
  "folderLink": "/secret/${torrent.downloadDir}/",
  "folderLinkEnabled": false,
  "incompleteFileLink": "/secret/${session.incomplete-dir}/",
  "incompleteFileLinkEnabled": false,
  "incompleteFolderLink": "/secret/${session.incomplete-dir}/",
  "incompleteFolderLinkEnabled": false,
  "labelMaticEnabled": false,
  "labelMaticRules": [
    [ "name", [ "Indiana.Jones" ], [ "Action", "Adventure" ] ],
    [ "downloadDir", [ "anime" ], [ "Animation", "Anime" ] ]
  ],
  "labels": [
    "Action", "Adventure", "Animation", "Anime", "Art", "Book", "Comedy",
    "Comic", "Crime", "Drama", "Detective", "Fantasy", "Food", "Game",
    "Gameshow", "History", "Horror", "Movie", "Music", "Musical", "Nature",
    "Romance", "Sci-fi", "Software", "Sport", "Talkshow", "Thriller", "TV",
    "War", "Western", "XXX"
  ],
  "localStorageEnabled": false,
  "mimeTypeIconTemplate": "https://raw.githubusercontent.com/eagerterrier/MimeTypes-Link-Icons/master/images/${file}-icon-64x64.png",
  "minSeeders": 8,
  "notificationLevel": 0,
  "pasteMagnet": false,
  "paths": [],
  "pathsPersistent": false,
  "registerMagnetHandler": false,
  "replaceMagnet": true,
  "screenshotMode": false,
  "showExitMessage": false,
  "showSpeedTitle": false,
  "soundEnabled": false,
  "soundDone": "",
  "soundVolume": 0.1,
  "styleSheet": [ "shift.css", "base.css", "terminal.css", "touch.css" ],
  "torrentLink": "/secret/torrents/",
  "torrentLinkEnabled": false,
  "torrentLinkTemplate": "wget \"${shift.torrentLink}${torrent.hashString}.torrent\" -O \"${torrent.name}.torrent\"",
  "torrentLinkTemplateEnabled": false,
  "trackers": "udp://tracker.torrent.eu.org:451/announce"
}
```

**Comment by Darkangeel-hd** [comment]
I'm having the same problem on my side

the 409 the server is sending complains about the header `X-Transmission-Session-Id`  needed to be updated
![image](https://github.com/user-attachments/assets/e572117d-4d14-41f2-8275-03c2d3995cbf)
which is done in the following request
![image](https://github.com/user-attachments/assets/279d709e-9887-4238-8906-343ae7667207)

I couldn't see any `paths` key on any of the rpc calls though

**Comment by killemov** [comment]
@Gladuin There was an untraceable issue before from another user and that eventually lead to a plugin messing with content. Could you try it without any plugins active? (Also I'm afraid I can't help you any further.)

@Darkangeel-hd The 409 error is standard and meant for CSRF protection. See https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md#231-csrf-protection

**Comment by Gladuin** [comment]
I'm not quite sure what you mean by plugins. I've installed transmission by just compiling 4.0.6 from source and running make install, nothing else apart from a few config changes (as you can see in my first post).

If you mean browser plugins, I just tried a fresh firefox install and it still gives an error upon trying to open the shift settings menu (although this time it's different from my last screenshot).

![image](https://github.com/user-attachments/assets/074a3b82-35fc-4dfd-9904-fc82662bba6b)

It's alright if you can't offer any further assistance, I'll just try to figure it out by myself since I assume it's just a stupid config error or something (because my setup is completely stock)

**Comment by killemov** [comment]
Yes, I meant browser plugins. You could try setting a breakpoint here: https://github.com/killemov/Shift/blob/3be3acb2f3b15359267ba93015bee1b6c95d90ea/shift.js#L6222
and then force reload the page.

**Comment by killemov** [comment]
@Gladuin Can you update and check again? There was a problem with handling a non-existent cookie.

**Comment by Gladuin** [comment]
@killemov Hi! I'm so sorry... I saw your previous comment at work and completely forgot about it.
I updated to the latest version of Shift (I just cloned the git repo) and it works perfectly with transmission 4.0.6! Seems like the cookie was indeed causing my issue.

**Comment by killemov** [comment]
Great!
